addons/l10n_ch/models/ir_actions_report.py

```diff
@@ -66,14 +66,23 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                         },
                         res_ids=qr_inv_ids,
                     )
+                    # Reassign QR codes to the appropriate index in case of multiple invoices
+                    if len(qr_res) > 1:
+                        stream_qr_pdf_combine = [value for value in qr_res.values() if value['stream'] is not None][0]
+                        for i in range(len(qr_res)-1):
+                            attachment_writer = OdooPdfFileWriter()
+                            attachment_writer.addPage(OdooPdfFileReader(stream_qr_pdf_combine['stream']).getPage(i))
+                            stream = io.BytesIO()
+                            attachment_writer.write(stream)
+                            qr_res[qr_inv_ids[i]]['stream'] = stream
 
                     for invoice_id, stream in qr_res.items():
-                        qr_pdf = OdooPdfFileReader(stream['stream'], strict=False)
```

Comment:
I think you should keep it as odoo to keep diff minimal (except the `if not invoice_id` part)